### ISSUE TITLE
Pnpm does not installed peerDependences and will not errors

### ISSUE THREAD

**Issue by zaiMoe** [pnpm/pnpm#6334]
### explain
In the pnpm workspace project：
```
apps/
	project-a/
		package.json
	project-b/
		package.json
package.json
```

The dependencies are as follows：
```
project-a/package.json： 
	dependencies:
		@my-pkg/pkg-b: ^1.0.0

project-b/package.json： 
	dependencies:
		@my-pkg/pkg-b: ^1.1.0
```

@my-pkg/pkg-table depends on the following：
```
@my-pkg/pkg-table/package.json： 
	peerDependencies
		@my-pkg/cell-render: ^1.0.0
		@my-pkg/action: ^1.0.0
	devDependencies
		@my-pkg/cell-render: ^1.0.0
		@my-pkg/action: ^1.0.0
```

### pnpm version: 
6.35.1

### Code to reproduce the issue:
```
1. pnpm install 
2. pnpm add @my-pkg/pkg-table @my-pkg/cell-render @my-pkg/action  --filter project-a
3. pnpm add @my-pkg/pkg-table --filter project-b
4. pnpm build --filter project-b
```
### Expected behavior:
Build reported an error because I did not install `@my-pkg/pkg-table` and `@my-pkg/cell-render` in project-b

### Actual behavior:
Build success!
`node_modules/.pnpm/`:
```
@my-pkg+pkg-table@1.0.0_hash1
    node_modules/
        @my-pkg/
            pkg-table
            cell-render   -> ../../../@my-pkg+cell-render@1.0.0_hash3
            action           -> ../../../@my-pkg+action@1.0.0_hash4
@my-pkg+pkg-table@1.0.0_hash2
    node_modules/
        @my-pkg/
            pkg-table
@my-pkg+cell-render@1.0.0_hash3
    node_modules/
        @my-pkg/
            cell-render
@my-pkg+action@1.0.0_hash4
    node_modules/
        @my-pkg/
            action
```

### Additional information:

 - `node -v` prints: v14.18.2
 - Windows, macOS, or Linux?: macOS

**Comment by zaiMoe** [comment]
> You should try with pnpm v8. pnpm v6 is not supported.

Yes, I know, but I want to know if this issue is just a bug in v6，I don't understand how it works.

**Comment by zkochan** [comment]
Yes, you are right. A warning should be printed when you run `pnpm add @my-pkg/pkg-table --filter project-b` unless the peer dependencies are optional peer dependencies.

However, with pnpm v8 the peer dependencies will be installed automatically, so there will be no warning (unless you set `auto-install-peers=false`.

**Comment by zaiMoe** [comment]
> Yes, you are right. A warning should be printed when you run `pnpm add @my-pkg/pkg-table --filter project-b` unless the peer dependencies are optional peer dependencies.
> 
> However, with pnpm v8 the peer dependencies will be installed automatically, so there will be no warning (unless you set `auto-install-peers=false`.

@zkochan  Thank you, zkochan. I will try to use pnnpm v8 in the future. There were warnings when `pnpm add @my-pkg/pkg-table --filter project-b`, but I couldn't find the dependencies referenced in project-b, making compilation error free in pnpm v6（no set `auto-install-peers=false`）

**Comment by zkochan** [comment]
Closing for lack of a reproduction on a supported version.

There is not enough information here to reproduce the problem, and there has been no follow-up.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
